core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Util.java

```diff
@@ -350,6 +350,7 @@ static Long expiresAtMillis(String token) {
 
   /** Class to handle authorization headers and token refresh. */
   public static class AuthSession {
+    private static int tokenRefreshNumRetries = 5;
```

Comment:
Can we use a constant name so people don't try to set it? `REFRESH_NUM_RETRIES` seems more appropriate.

Reply:
unfortunately no as this is being prevented by our checkstyle rules to name something all uppercase when it's not a `static final`